documentation/swagger-checklist.md

```diff
@@ -43,82 +45,236 @@ This document defines a Swagger checklist for Microsoft Azure, along with clarif
+- [ ] **M2003**: If a format is specified along with a type, it *MUST* be from one of the supported formats - "int64", "int32", "float", "decimal", "double", "byte", "base64url", "unixtime", "date", "date-time", "duration", "date-time-rfc1123", "uuid", otherwise in codegeneration the artifact will be of the specified type and not as per the desired format. 
 
-&#x25a2; *M2005*: A collection of references that are read only MUST be marked as read only in Swagger.
+- [ ] **M2004**: Long running (asynchronous) operations *MUST* be modeled as [long running operations](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#longrunning) in Swagger.
 
-&#x25a2; *M2006*: Mutability of properties MUST be marked up using [`x-ms-mutability` extension](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-mutability).
+- [ ] **M2005**: A longrunning operation *MUST* have a terminal success status code. At least 
+   - 200 or 201 for PUT/PATCH
+   - 200, 201 or 204  or all of the aforementioned for POST
+   - 204 or 200 or both for delete. 
+   This will indicate the code generator, how to deserialize the final success response after the polling is done. If the response has a response body then it *MUST* be modeled in the terminal status code.
 
-&#x25a2; *M2007*: Error definitions MUST be abstracted, modeled and reused instead of introducing a plethora of error definitions that look the same with different names.
+- [ ] **M2005**: Required parameters *MUST* be accurately labeled as `"required": true` in Swagger.
 
-&#x25a2; *M2008*: A Swagger spec MUST NOT define operations or properties or parameters for functionalities that MAY work in the future.
+- [ ] **M2006**: Required properties of a model definition *MUST* be accurately labeled as `"required": [ "propertyName1" ]` in Swagger.
 
-For example, `OutBoundNATRule` is a model that exists in a the networking Swagger but it is not supported by the Networking RP.
+- [ ] **M2007**: Mutability of properties *MUST* be marked up using [`x-ms-mutability` extension](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-mutability).
 
-&#x25a2; *M2009*: A collection model in Swagger MUST support adding more than one element to the collection.
+- [ ] **M2008**: Error definitions *MUST* be abstracted, modeled and reused.
 
-&#x25a2; *M2010*: Operations that are exposed through private previews MUST not be described in Swaggers in the [public GitHub repository](https://github.com/Azure/azure-rest-api-specs).
+- [ ] **M2009**: A Swagger spec *MUST NOT* define operations or properties or parameters for functionalities that are not currently supported by the service in the given api-version. The primary goal of the spec is that it *MUST* correctly and completely represent the underlying REST API.
 
+- [ ] **M2010**: A collection model in Swagger *MUST* support adding more than one element to the collection. TODO: Ask for clarity.
```

Comment:
if the service is doing it like that and if they are not willing to change it then this will result in an error and we wont accept the PR. I think this must be in the SHOULD category. It is ok for a service to build a forward looking api. 
one would want to weigh breaking changes in the structure (object -> array) v/s a requirement like this